bulk_indexer_test.go

```diff
@@ -295,3 +297,77 @@ func TestItemRequireDataStream(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, int64(2), stat.Indexed)
 }
+
+func TestPopulateFailedItemSource(t *testing.T) {
```

Comment:
Should we add a negative check, or is that covered elsewhere? i.e. check that Source is not populated unless we set the new config